[EDU-3730] feat: add customize table grafana (#598)
aziontech/docs: src/content/docs/en/pages/observe-journey/real-time-events/integrations/grafana-plugin-custom-table.mdx

---
title: How to customize a log table with the Azion plugin on Grafana

> description: The Azion data source plugin on Grafana allows you visualize the data from your existing applications on Azion on a Grafana dashboard.

Reviewer: ```suggestion
description: The Azion data source plugin on Grafana allows you to visualize the data from your Azion edge applications on a Grafana dashboard.
```

meta_tags: azion, edge, observe, observability, logs, grafana, analytics
namespace: docs_plugin_grafana_log_table
permalink: /documentation/products/guides/observe/azion-plugin-grafana-customize-log-table/
---

import Button from '~/components/Button.astro'

With the Azion data source plugin on Grafana, you can customize your own dashboard to visualize the data you need.

<Button href="/en/documentation/products/guides/observe/integrate-grafana/" text="go to integrate with grafana guide" variant="secondary" target="_blank"></Button>

---

## Customizing a dashboard on Grafana

After [installing a Grafana instance locally](https://github.com/aziontech/grafana-plugin/blob/dev/README.md#install-azion-plugin-on-local-grafana-install-locally) on your machine and authorizing the Azion plugin, open it.

Follow the next steps:

> 1. On the left-side menu, click on **Dashboards**. The dashboard page opens.

Reviewer: ```suggestion
1. On the left-side menu, click on **Dashboards** to open the dashboard page.
```

2. Next to the search bar, click **New** > **New Dashboard**.
3. Select **Add visualization**.
4. On the second section of the page, below the preview, on **Data source** dropdown menu, select **Azion**.
5. On the code box, add the query you want to use.
    - See the [GraphQL API guides](/en/documentation/products/graphql-api-overview) for a few examples of queries.

The five fields below can be used to complete your setup depending on what type of visualization you're using and what you want to see on your graphs:

> - **Data path**: insert the [data set you want to use](/en/documentation/products/graphql-api-features/#datasets).

Author: ```suggestion
- **Data path**: add the value **Events**.
```

    - To analyze raw logs with a large amount of data or top X queries, it's recommended to use the **Table** type. You can combine it with other type of charts and graphs.
7. On the upper-right corner, click **Save** to apply your configurations.

---

## Read more on Grafana

- [Create and use dashboards](https://grafana.com/docs/grafana/latest/dashboards/)
- [Grafana dashboard best practices](https://grafana.com/docs/grafana/latest/dashboards/build-dashboards/best-practices/)
- [Add and manage variables](https://grafana.com/docs/grafana/latest/dashboards/variables/)
- [Configuring time series in ISO8601](https://momentjs.com/docs/#/parsing/string/)
- [Configuring time series in custom format](https://momentjs.com/docs/#/parsing/string-format/)
- [Visualizations](https://grafana.com/docs/grafana/latest/panels-visualizations/visualizations/)
- [Configuring a legend](https://grafana.com/docs/grafana/latest/panels-visualizations/visualizations/configure-legend/)
- [Configuring value mappings](https://grafana.com/docs/grafana/latest/panels-visualizations/configure-value-mappings/)

---

### Trademarks

> [Grafana Cloud](https://grafana.com/products/cloud/) is a trademark of Grafana Labs. We are not affiliated with, endorsed or sponsored by Grafana Labs or its affiliates.

Reviewer: ```suggestion
[Grafana Cloud](https://grafana.com/products/cloud/) is a trademark of Grafana Labs. We aren't affiliated with, endorsed, or sponsored by Grafana Labs or its affiliates.
```
